### PR TITLE
[UPGRADE] Fixes for #128 and #129

### LINKED ISSUE #128: Improving test utility checkCovCommand
**Description**

The `checkCovCommand` utility function currently uses a mutable default argument `args=[None]`. This is considered a bad practice in Python because mutable default arguments can lead to unexpected behavior due to the way Python handles default parameter values.

**Plan to Refactor**

1. **Modify the Function Signature**: Change the default value of `args` to `None` in the function definition to avoid using a mutable default argument.

2. **Initialize Within the Function**: Inside the function, check if `args` is `None` and initialize it accordingly.

3. **Update Dependent Code**: Review and update any code that calls `checkCovCommand` to ensure compatibility with the new function signature.

4. **Implement Test Cases**: Write test cases for different scenarios to ensure proper behavior:
   - `args=[]`
   - `args=None`
   - `args='not an array at all'`

5. **Verify Functionality**: Run existing tests to confirm that the refactored function does not break any current functionality and handles all input types as expected.

**References**

- PR: https://github.com/reactive-firewall/multicast/pull/127
- Comment: https://github.com/reactive-firewall/multicast/pull/127#discussion_r1774288383
- Requested by: @reactive-firewall

### LINKED ISSUE #129: test utility refactoring for time/fuzzing wrappers of checkPythonCommand
### Summary

In the discussion on PR #127 (https://github.com/reactive-firewall/multicast/pull/127) (see comment https://github.com/reactive-firewall/multicast/pull/127#discussion_r1774288385 by @reactive-firewall), we agreed to refactor the `timePythonCommand` and `checkPythonFuzzing` functions to use `*args` and keyword-only arguments. This change avoids mutable default arguments, aligns with the intended functionality, and improves debugging.

### Plan for Next Steps

- **Refactoring:**
  - Update the `timePythonCommand` and `checkPythonFuzzing` functions to use `*args` and keyword-only arguments (`stderr=None`).
  - Modify the functions to raise a `RuntimeError` if `args` is empty or the first argument is `None`.
  - Ensure that `checkPythonCommand` can handle `args` as a tuple or convert it to a list if necessary.

- **Testing:**
  - Write new unit tests to cover the updated functions, including tests for the new exception handling when `args` is empty or contains `None`.
  - Run the test suite to ensure that all tests pass with the refactored code.

- **Documentation:**
  - Update the docstrings of the refactored functions to reflect the new parameter structure and behavior.
  - Document any changes in the project's documentation, highlighting the refactoring and its benefits.

### References

- PR #127: https://github.com/reactive-firewall/multicast/pull/127
- Comment: https://github.com/reactive-firewall/multicast/pull/127#discussion_r1774288385 by @reactive-firewall

### PR DESCRIPTION
**Pull request by reactive-firewall**
- [x] Closes #128 
- [x] Closes #129

Changes in file tests/context.py:
 - `checkCovCommand()`: possible solution for PYL-W0102
 - `checkPythonCommand()`: possible solution for PLY-W0102
 - refactored time/fuzz utilities to ignore any default, to keep code compact (- WIP # 129 -)

---

## Summary by CodeRabbit

- **Bug Fixes**
	- Enhanced argument handling in command functions to prevent errors when no arguments are provided.
- **New Features**
	- Improved flexibility in passing command arguments through variable-length argument lists.

### CODE REVIEW

File: tests/context.py [hunk 1]
```diff
@@ -468,38 +472,61 @@ def checkPythonCommand(args, stderr=None):
...
+
+	Args:
+		args (array): An array of positional command arguments to be executed.
+		stderr (Optional[int]): File descriptor for stderr redirection.
+		Defaults to None.
+
+		Returns:
+			The output of checkPythonCommand.
+	"""
+	return checkPythonCommand(args, stderr=stderr)
```
Comment (reactive-firewall) [inline tests/context.py:501]: @coderabbitai, Please read through all the python files in the project repo, GH issue #128, and GH issue #129, then generate some (about 4 to 6) relevant doctests for the refactored function `timePythonCommand` around line 489, using a style and tone consistent to that of the rest of the doctests (including the naming convention in `tests/` of "Meta Testing:" for doctests of test-code etc.)

File: tests/context.py [hunk 2]
```diff
@@ -468,38 +472,61 @@ def checkPythonCommand(args, stderr=None):
...
+		Defaults to None.
 
-def checkPythonFuzzing(args=[None], stderr=None):  # skipcq: PYL-W0102  - [] != [None]
-	"""function for backend subprocess check_output command"""
+	Returns:
+		str: The command output as a string.
+
+	Raises:
+		RuntimeError: If an error occurs during command execution.
+	"""
```
Comment (reactive-firewall) [inline tests/context.py:518]: @coderabbitai, Please read through all the python files in the project repo, GH issue #128, and GH issue #129, then generate some (about 4 to 6) relevant doctests for the refactored function `checkPythonFuzzing` around line 504, using a style and tone consistent to that of the rest of the doctests (including the naming convention in `tests/` of "Meta Testing:" etc.)

File: tests/context.py [hunk 3]
```diff
@@ -252,7 +252,7 @@ def getPythonCommand():
 	return str(thepython)
 
 
-def checkCovCommand(args=[None]):  # skipcq: PYL-W0102  - [] != [None]
+def checkCovCommand(*args):  # skipcq: PYL-W0102  - [] != [None]
```
Comment (reactive-firewall) [inline tests/context.py:255]: @coderabbitai, Please read through all the python files in the project repo, GH issue #128, and GH issue #129, then generate some (about 4 to 6) relevant doctests for the refactored function `checkCovCommand` around line 255, using a style and tone consistent to that of the rest of the doctests (including preserving any and all exsisting doctests)
